### PR TITLE
Update k8s 1.14

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

* Upgrade kuberntes vendors to v1.14.0-rc.1
* Upgrade go version to 1.12.1 (both aks-engine and kubernetes have been upgraded to 1.12.1)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

**Comment by feiskyer** [comment]
e2e cases have been passed, but got another error:

```
main.go:307: Something went wrong: encountered 1 errors: [cannot get the delete resource group future response: Future#WaitForCompletion: context has been cancelled: StatusCode=202 -- Original Error: context deadline exceeded]
```

I think the timeout for resource group should be set to a bigger value, or we can just omit the deletion errors.

**Comment by feiskyer** [comment]
Run again and it passed now.

**Comment by ritazh** [comment]
@feiskyer I think even the latest released version of aks-engine only supports up to `1.14.0-beta.1` so to support `v1.14.0-rc.1` we have two options:
1. update https://github.com/kubernetes/cloud-provider-azure/blob/master/tests/k8s-azure/manifest/linux.json to use 
```yaml
customHyperkubeImage: "gcrio.azureedge.net/google_containers/hyperkube-amd64:v1.14.0-rc.1"

...

distro: "ubuntu"
```
2. wait for a released version of aks-engine that supports `v1.14.0-rc.1`

I think we are always going to be a bit ahead of aks-engine. I propose we go with option 1 so we can test more recent releases of k8s. Also the same suggested changes would need to be made in https://github.com/kubernetes/cloud-provider-azure/blob/master/tests/k8s-azure/manifest/linux-autoscaler.json for the autoscaler e2e tests.

WDYT?

**Comment by feiskyer** [comment]
Option 1 lgtm. we're actually always going ahead of aks-engine

**Comment by feiskyer** [comment]
@ritazh Addressed comments, PTAL

File: tests/k8s-azure/manifest/linux.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
             "orchestratorRelease": "1.14",
             "kubernetesConfig": {
                 "useCloudControllerManager": true,
-                "customCcmImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-alpha.1",
+                "customCcmImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-rc.1",
```
Comment (ritazh) [inline tests/k8s-azure/manifest/linux.json]: I think you forgot to add `customHyperkubeImage: "gcrio.azureedge.net/google_containers/hyperkube-amd64:v1.14.0-rc.1"`

File: tests/k8s-azure/manifest/linux-autoscaler.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
             "orchestratorRelease": "1.14",
             "kubernetesConfig": {
                 "useCloudControllerManager": true,
-                "customCcmImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-alpha.1",
+                "customCcmImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-rc.1",
```
Comment (ritazh) [inline tests/k8s-azure/manifest/linux-autoscaler.json]: pls add `customHyperkubeImage: "gcrio.azureedge.net/google_containers/hyperkube-amd64:v1.14.0-rc.1"`

File: tests/k8s-azure/manifest/linux.json (hunk 1 above)
Comment (feiskyer) [reply tests/k8s-azure/manifest/linux.json]: oops, added now.

File: tests/k8s-azure/manifest/linux-autoscaler.json (hunk 2 above)
Comment (feiskyer) [reply tests/k8s-azure/manifest/linux-autoscaler.json]: added now.

File: tests/k8s-azure/manifest/linux-autoscaler.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
...
+                "customHyperkubeImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-rc.1",
```
Comment (ritazh) [inline tests/k8s-azure/manifest/linux-autoscaler.json]: This should be `gcrio.azureedge.net/google_containers/hyperkube-amd64:v1.14.0-rc.1`

File: tests/k8s-azure/manifest/linux-kcm.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
             "orchestratorType": "Kubernetes",
             "orchestratorRelease": "1.14",
             "kubernetesConfig": {
-                "customHyperkubeImage": "gcrio.azureedge.net/google_containers/hyperkube-amd64:v1.13.0-alpha.3",
+                "customHyperkubeImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-rc.1",
```
Comment (ritazh) [inline tests/k8s-azure/manifest/linux-kcm.json]: same here

File: tests/k8s-azure/manifest/linux.json [hunk 5]
```diff
@@ -7,7 +7,8 @@
...
+                "customHyperkubeImage": "gcrio.azureedge.net/google_containers/cloud-controller-manager-amd64:v1.14.0-rc.1",
```
Comment (ritazh) [inline tests/k8s-azure/manifest/linux.json]: same

File: tests/k8s-azure/manifest/linux-autoscaler.json (hunk 3 above)
Comment (feiskyer) [reply tests/k8s-azure/manifest/linux-autoscaler.json]: fixed.

File: tests/k8s-azure/manifest/linux-kcm.json (hunk 4 above)
Comment (feiskyer) [reply tests/k8s-azure/manifest/linux-kcm.json]: fixed.

File: tests/k8s-azure/manifest/linux.json (hunk 5 above)
Comment (feiskyer) [reply tests/k8s-azure/manifest/linux.json]: fixed

**Comment by feiskyer** [comment]
The new job configs https://github.com/kubernetes/test-infra/pull/11846 have been merged and it's upgraded to v1.14.0-rc.1. And it's changing to be the required check now.

Because the failure of pull-cloud-provider-azure-e2e is actually bug from kubetest, let's merge this manually.

**Pull request merged**